portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,29 +36,44 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    UserService userService = UserServiceFactory.getUserService();
+    PrintWriter out = response.getWriter();
+    if (userService.isUserLoggedIn()) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+        String title = (String) entity.getProperty("title");
+        long timestamp = (long) entity.getProperty("timestamp");
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      String title = (String) entity.getProperty("title");
-      long timestamp = (long) entity.getProperty("timestamp");
+        Comment new_comment = new Comment( title, timestamp);
```

Comment:
You should rename the variable to be newComment, as local variable names in Java should be in camel case.